--- /usr/share/perl5/Net/DNS/ZoneFile/Fast.pm	2011-12-05 10:11:31.000000000 +0000
+++ /usr/share/perl5/Net/DNS/ZoneFile/Fast.pm.new	2011-12-05 10:11:24.000000000 +0000
@@ -604,6 +604,33 @@
 	  } else {
 	      error("bad txtdata in TXT");
 	  }
+      } elsif (/\G(type[0-9]+)[ \t]+/igc) {
+          my $type = $1;
+	  if (/\G\\#\s+(\d+)\s+\(\s(.*)$/gc) {
+	      # multi-line
+	      $sshfp = { 
+			Line    => $ln,
+			name    => $domain,
+			type    => uc $type,
+			ttl     => $ttl,
+			class   => "IN",
+                        fptype  => $1,
+			fingerprint => $2,
+		       };
+	      $parse = \&parse_sshfp;
+	  } elsif (/\G\\#\s+(\d+)\s+(.*)$pat_skip$/gc) {
+	      push @zone, {
+			   Line    => $ln,
+			   name    => $domain,
+			   type    => uc $type,
+			   ttl     => $ttl,
+			   class   => "IN",
+                           fptype  => $1,
+			   fingerprint => $2,
+			  };
+	  } else {
+	      error("bad data in in $type");
+	  }
       } elsif (/\G(sshfp)[ \t]+/igc) {
 	  if (/\G(\d+)\s+(\d+)\s+\(\s*$/gc) {
 	      # multi-line
@@ -945,7 +972,7 @@
       } elsif (/\Gany\s+tsig.*$/igc) {
 	  # XXX ignore tsigs
       } else {
-	  error("unrecognized type");
+	  error("unrecognized type for $domain\n$_\n");
       }
   }
 
